src/Middleware/HttpLogging/src/W3CLoggerProcessor.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class W3CLoggerProcessor : IDisposable
+    {
+        private const int _maxQueuedMessages = 1024;
+
+        private readonly string _path;
+        private readonly string _fileName;
+        private readonly int? _maxFileSize;
+        private readonly W3CLoggingFields _loggingFields;
+        private bool _hasWritten;
+
+        private readonly BlockingCollection<string> _messageQueue = new BlockingCollection<string>(_maxQueuedMessages);
+        private Task _outputTask;
+
+        public W3CLoggerProcessor(IOptionsMonitor<W3CLoggerOptions> options)
+        {
+            var loggerOptions = options.CurrentValue;
+            _path = loggerOptions.LogDirectory;
+            _fileName = loggerOptions.FileName;
+            _maxFileSize = loggerOptions.FileSizeLimit;
+            _loggingFields = loggerOptions.LoggingFields;
+
+            FullName = Path.Combine(_path, $"{_fileName}{Guid.NewGuid().ToString()}.log");
```

Comment:
Let's use UTC timestamps, not GUIDs here

Reply:
I'd look at ANCM logs for prior art. I remember customers complaining that they wanted a stable sort for their file names.